src/Ethereum/MessageSigner.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "HexCoding.h"
+#include "MessageSigner.h"
+#include <Ethereum/ABI/ParamStruct.h>
+#include <nlohmann/json.hpp>
+#include <sstream>
+
+namespace TW::Ethereum::internal {
+
+Data generateMessage(const std::string& message) {
+    std::string prefix(1, MessageSigner::EthereumPrefix);
+    std::stringstream ss;
+    ss << prefix << MessageSigner::MessagePrefix << std::to_string(message.size()) << message;
+    Data signableMessage = Hash::keccak256(data(ss.str()));
+    return signableMessage;
+}
+
+std::string commonSign(const PrivateKey& privateKey, const Data& signableMessage, MessageType msgType, TW::Ethereum::MessageSigner::MaybeChainId chainId) {
+    auto data = privateKey.sign(signableMessage, TWCurveSECP256k1);
+    switch (msgType) {
+    case MessageType::ImmutableX:
+        break;
+    case MessageType::Legacy:
+        data[64] += 27;
+        break;
+    case MessageType::Eip155:
+        auto id = chainId.value_or(0);
+        data[64] += 35 + id * 2;
```

Comment:
nice magic numbers :)

Reply:
Those magic numbers were not wrapped in the code, and there is no explanation here: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md it's unprecise how we should name them